msgpack/fallback.py

```diff
@@ -696,7 +696,7 @@ def _pack(self, obj, nest_limit=DEFAULT_RECURSE_LIMIT,
                 else:
                     raise PackValueError("Bytes is too large")
                 return self._buffer.write(obj)
-            if check(obj, (Unicode, bytes, memoryview)):
+            if check(obj, (Unicode, bytes)):
```

Comment:
the easiest way to fix your code, is to keep memoryview in that list.
and move the block you added below to above this block, using
```
if self._use_bin_type and check(obj, memoryview):
```